src/middlewares.js

```diff
@@ -23,6 +23,41 @@ const getMountForRequest = function (req) {
   return req.protocol + '://' + req.get('host') + mountPath;
 };
 
+const checkIpRanges = (ip, ranges = []) => {
+  if (!ip) {
+    return false;
+  }
+  const transformIp = ip => {
```

Comment:
Why do we need this?

This string parsing may not be extensive enough. I think you could omit leading zeros partially and use for example `::0000:1` instead of `::1`.

Reply:
Which bit of code are you referring to?

`if (!ip)` is for certain tests that do not provide an IP

`transformIp` is to allow `::1` as a shorthand for `127.0.0.1`

https://github.com/whitequark/ipaddr.js/issues/162

Reply:
So there’s no need for the conversion?

Reply:
> by default nodejs treats ::, ::1, and 127.0.0.1 as seperate addresses. 

They are in fact different addresses, so that behavior would be expected.

> The additional code just interprets them all as localhost

It's up to the developer to properly set the parse server config, depending on the socket properties. In theory one could run 2 different Parse Servers, one on `127.0.0.1`, the other one on `::1`.

- `127.0.0.1` for localhost on an IPv4 socket
- `::1` for localhost on an IPv6 socket
- `::` we could interpret as `::1` (not as `127.0.0.1`), but we don't have to since it's outside RFC

So I don't think we need any conversion.